Evolutionary_loop/Archived/ssPermissionCheck.py

```diff
@@ -0,0 +1,53 @@
+#*******************************************************************************
+#  file: ssPermissionCheck.py
+#     This script is based on ssPermissionCheck.sh
+#     This is an attempt to slowly migrate to Python.
+#
+#  Programmer: Jason Yao (yao.966@osu.edu)
+#
+#  Revision history:
+#     07/18/23  Original version
+#
+#  Notes:
+#     * vertical ruler at column 80
+#
+#  TODO: 
+#     * ssPermissionCheck.sh seems capable of waiting (sleep 1m);
+#       add this in the future.
+#     * ssPermissionCheck.sh sends email with error messages; add this.
+#
+#*******************************************************************************
+import argparse as ap
+
+### ARGUMENTS ###
+# Use the following command to checkout the arguments in the terminal
+# >> Python3 ssPermissionCheck.py -h
+parser = ap.ArgumentParser()
+parser.add_argument("RunName", help="Run Name")
+parser.add_argument("WorkingDir", help="Working Directory (Evolutionary_Loop)")
+parser.add_argument("usr_spec", help="User-specified checkpoint (aka state)")
+g = parser.parse_args()
+
+
+### MESSAGES ###
+congrats = \
+"Hurray! Checkpoint (line 2) in SaveState.txt matches what you specified"
+err = \
+"SaveState did not update after jobs finished.\
+ Permissions were likely not opened after the last run."
+
+
+### READING ###
+# The "f" below stands for "f-strings".
+# This is cleaner than the old way of using %'s for string interpolation.
+file_path = f'{g.WorkingDir}/SaveStates/{g.RunName}_SaveState.txt'
+
+file = open(file_path,"r")    # Open the file (read mode)
+file.readline()               # reading the first line (generation); not needed
+checkpoint = file.readline()  # reading the second line (state)
+
+### MAIN ###
```

Comment:
A common pattern is to have something like:
```py
def main(args):
    ...

if __name__ == '__main__':
    # argparse stuff
    main(args)
```
The if at the bottom is executed only if the module is executed directly with `python module.py` but not if it's imported by other modules as `import module`.  Having a `main` function also allows other modules to still perform the operations (e.g. the permission check).

If this file is only intended to be executed directly and not to be imported, then you don't need all this and what you did is ok.

Reply:
I've changed the file accordingly for practice. Is the way I'm doing it right?
`args` for `main` in this case would be empty, right?